### PR TITLE
Add Check and Change Password for an Account

### LINKED ISSUE #2389: Allow user to validate & reset password for account (pre-sending, non-Signer)
- execute eth_sign with password to validate

(from @gavofyork)

### PR DESCRIPTION
**Pull request by ngotchac**
Closes #2389 

Adds a new modal in the Account view to check if a password is correct, and to modify it.
Also adds the corresponding endpoints to the JS API.

### CODE REVIEW

**Comment by jacogr** [comment]
Nice! Picture? ;)

**Comment by jacogr** [comment]
Overall comments -

1. Move the password button to last on the list on the ActionBar (after Edit)

2. We need the ability to update the password Hint along with a password Change

3. The "Test Password" & "Change Password" headings are not intuitive, i.e. on first glance no idea that there are options behind them (We only have this concept on swaps in the Settings menus, so needs a bit of work here to make it look tab-y)

File: js/src/modals/PasswordManager/passwordManager.js [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    repeatNewPass: '',
+    repeatValid: true
+  }
+
+  render () {
+    return (
+      <Modal
+        actions={ this.renderDialogActions() }
+        title='Password Manager'
+        wa
```
Comment (jacogr) [inline js/src/modals/PasswordManager/passwordManager.js]: What is `wa`?

Comment (ngotchac) [inline js/src/modals/PasswordManager/passwordManager.js]: A mistake ;)

**Comment by ngotchac** [comment]
@jacogr Okay it should be better now :+1:

**Pull request merged**